### PR TITLE
adding test case for bug in es6 destructuring

### PR DESCRIPTION
**Pull request by Pandeyjidev**

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -120,9 +120,42 @@ export function functionDocString ( signature: string ): string {
  */
 export function extractParameters ( signature: string ): string[] {
     const possibleParameters: string = signature.slice( signature.indexOf( '(' ) + 1, signature.indexOf( ')' ) ).trim();
-    return (possibleParameters === '') ? [] : possibleParameters.split( ',' ).map( (str: string) => str.trim() );
+    console.log('possibleParameters ', possibleParameters);
```
Comment (mohseenrm) [inline src/extension.ts]: Remove this statement

File: src/extension.ts [hunk 2]
```diff
@@ -120,9 +120,42 @@ export function functionDocString ( signature: string ): string {
...
+    const filteredArguments: string[] = (possibleParameters === '') ? [] : possibleParameters.split( ',' ).map( (str: string) => str.trim() );
```
Comment (mohseenrm) [inline src/extension.ts]: spacing

File: src/extension.ts [hunk 3]
```diff
@@ -120,9 +120,42 @@ export function functionDocString ( signature: string ): string {
...
+    const filteredArgumentsReturned: string[] = filteredArguments.map(mapES6Destructuring);
```
Comment (mohseenrm) [inline src/extension.ts]: spacing

File: src/extension.ts [hunk 4]
```diff
@@ -120,9 +120,42 @@ export function functionDocString ( signature: string ): string {
...
+    return filteredArgumentsReturned;
+}
+function mapES6Destructuring (arg: string): string {
```
Comment (mohseenrm) [inline src/extension.ts]: spacing

File: test/extension.test.ts [hunk 5]
```diff
@@ -51,6 +51,18 @@ suite('Doc String generation', () => {
...
     });
 
+    test.only('check parameters with es6 destructuring', () => {
+        const signature: string = 'const connect = ( url , { USE_SSL , SSL , logger = console  }) => {';
+        const signature2: string = 'function( { firstparam , secondparam, thirdparam}, y, z)';
+        const signature3: string = 'function( x, { firstparam , secondparam = 123, thirdparam}, z)';
+        const signature4: string = '( firstparam = 123 , { secondparam, thirdparam}, y, z)';
+
+        expect( extractParameters( signature ) ).to.deep.equal( ['url', 'USE_SSL', 'SSL', 'logger'] );
+        expect( extractParameters( signature2 ) ).to.deep.equal( ['firstparam', 'secondparam', 'thirdparam', 'y', 'z'] );
```
Comment (mohseenrm) [inline test/extension.test.ts]: spacing

**Review by mohseenrm — CHANGES_REQUESTED** [review]
styling bugs

**Comment by mohseenrm** [comment]
Good job! @Pandeyjidev

**Review by danielo515 — COMMENTED** [review]
Good job. But in my opinion if a default parameter is provided then the property should be flagged as optional and document the default parameter.
Regards

**Pull request merged**